packages/parser-less/src/parser.ts

```diff
@@ -0,0 +1,55 @@
+const parser = require('postcss-less');
+
+import * as postcss from 'postcss';
+
+import { debug as d } from '@hint/utils/dist/src/debug';
+import { normalizeString } from '@hint/utils/dist/src/misc/normalize-string';
+import { HTMLElement } from '@hint/utils/dist/src/dom/html';
+import { Parser } from 'hint/dist/src/lib/types';
+import { StyleEvents } from '@hint/parser-css';
+import { Engine } from 'hint';
+
+const debug = d(__filename);
+
+export default class CSSParser extends Parser<StyleEvents> {
+    public constructor(engine: Engine<StyleEvents>) {
+        super(engine, 'less');
+
+        const emitLESS = async (code: string, resource: string, element: HTMLElement | null) => {
+
+            try {
+                await this.engine.emitAsync(`parse::start::css`, { resource });
+
+                const result = await postcss().process(code, { from: resource, parser });
+                const ast = result.root!; // always defined even for '' (typings error?)
+
+                await this.engine.emitAsync(`parse::end::css`, {
+                    ast,
+                    code,
+                    element,
+                    resource
+                });
+
+            } catch (err) /* istanbul ignore next */ {
+                debug(`Error parsing LESS code: ${code} - ${err}`);
+            }
+        };
+
+        engine.on('fetch::end::*', async (fetchEnd) => {
+            const code = fetchEnd.response.body.content;
+            const resource = fetchEnd.resource;
+
+            if (fetchEnd.response.mediaType === 'text/less') {
+                await emitLESS(code, resource, null);
+            }
+        });
+
+        engine.on('element::style', async ({ element, resource }) => {
+            const rel = normalizeString(element.getAttribute('rel'));
```

Comment:
I'm looking at the [docs and `rel` is when using a `.less` file](http://lesscss.org/usage/#using-less-in-the-browser):

```html
<link rel="stylesheet/less" type="text/css" href="styles.less" />
```

so this won't trigger with the current event you are subscribing to.

The docs doesn't seem to document it, but it looks like any `<style type="">` that matches `/^text\/(x-)?less$/` should work:

https://github.com/less/less.js/blob/cae5021358a5fca932c32ed071f652403d07def8/lib/less-browser/index.js#L64

Reply:
`<link>`, `<style>`, what's the difference?🤦‍♂️

I've also been wondering if `<style lang="less">` works with Vue since that seems to be the approach there for single file components once we're ready to analyze them. I see examples for SASS, but haven't found anything concrete for LESS yet.